src/EditorFeatures/CSharpTest/AddParameter/AddParameterTests.cs

```diff
@@ -584,5 +584,1027 @@ void M()
+    }
+}");
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocationLocalFunction()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+class C1
+{
+    void M1()
+    {
+        int Local() => 1;
+        [|Local|](2);
+    }
+}",
+@"
+class C1
+{
+    void M1()
+    {
+        int Local(int v) => 1;
+        Local(2);
+    }
+}");
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        [Trait("TODO", "Fix broken")]
+        public async Task TestInvocationLambda1()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+using System;
+class C1
+{
+    void M1()
+    {
+        Action a = () => { };
+        [|a|](2);
+    }
+}",
+@"
+using System;
+class C1
+{
+    void M1()
+    {
+        Action a = (int v) => { };
+        a(2);
+    }
+}");
+            //Should be Action<int> a = (int v) => { };
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocationStaticMethod()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+class C1
+{
+    static void M1()
+    {
+    }
+    void M2()
+    {
+        [|M1|](1);
+    }
+}",
+@"
+class C1
+{
+    static void M1(int v)
+    {
+    }
+    void M2()
+    {
+        M1(1);
+    }
+}");
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocationExtensionMethod()
+        {
+            var code =
+@"
+namespace N {
+static class Extensions
+{
+    public static void ExtensionM1(this object o)
+    {
+    }
+}
+class C1
+{
+    void M1()
+    {
+        new object().[|ExtensionM1|](1);
+    }
+}}";
+            var fix =
+@"
+namespace N {
+static class Extensions
+{
+    public static void ExtensionM1(this object o, int v)
+    {
+    }
+}
+class C1
+{
+    void M1()
+    {
+        new object().ExtensionM1(1);
+    }
+}}";
+            await TestInRegularAndScriptAsync(code, fix);
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocationOverride()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+class Base
+{
+    protected virtual void M1() { }
+}
+class C1 : Base
+{
+    protected override void M1() { }
+    void M2()
+    {
+        [|M1|](1);
+    }
+}",
+@"
+class Base
+{
+    protected virtual void M1() { }
+}
+class C1 : Base
+{
+    protected override void M1(int v) { }
+    void M2()
+    {
+        M1(1);
```

Comment:
virtual was not updated as well.

Reply:
Yes. I have a section in one of the comments that discusses this kind of problems (*Interface inheritance and method overriding*). I proposed to not fix along the inheritance chain but that might not be what you want.

Reply:
is there a reason this is provided as one of option? it doesn't seems reducing much work from users.

Reply:
tagging @kuhlenh @jinujoseph since it basically makes override to have 2 entries in code fix list.